cms/articles/forms.py

```diff
@@ -0,0 +1,45 @@
+from secrets import token_urlsafe
+from typing import Any
+
+from wagtail.blocks.stream_block import StreamValue
+from wagtail.models import Page
+
+from cms.core.forms import PageWithCorrectionsAdminForm
+from cms.taxonomy.forms import DeduplicateTopicsAdminForm
+
+
+class PageWithHeadlineFiguresAdminForm(DeduplicateTopicsAdminForm):
+    def __init__(
+        self,
+        *args: Any,
+        parent_page: Page | None = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        # Save a reference to the parent page which we use in the clean method
+        self.parent_page = parent_page
+
+    def clean_headline_figures(self) -> StreamValue:
+        headline_figures: StreamValue = self.cleaned_data["headline_figures"]
+        if not headline_figures:
+            # No headline figures, so return an empty stream value
+            return headline_figures
+
+        # Check if editing a new page
+        if self.instance.pk is None and self.parent_page:
+            # Grab the latest page in the series
+            latest = self.parent_page.get_latest()
+            if latest:
+                # The figures were inherited, so we need to also inherit the figure ids
```

Comment:
It is clear now, but in a few months the inheritance mechanism might not be obvious, worth changing to something like: 
```suggestion
                # The figures were inherited via the init_new_page signal, so we need to also inherit the figure ids
```

Reply:
722370e